segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -167,6 +175,59 @@ public void setupAppend(SetupAppend setupAppend) {
                 });
     }
 
+    @VisibleForTesting
+    CompletableFuture<Void> setupTokenExpiryTask(SetupAppend setupAppend) {
+        String segment = setupAppend.getSegment();
+        UUID writerId = setupAppend.getWriterId();
+        long requestId = setupAppend.getRequestId();
+
+        Duration expiryDuration = this.durationToExpiry(setupAppend);
+
+        if (expiryDuration == null) {
+            return CompletableFuture.completedFuture(null);
+        } else {
+            return Futures.delayedTask(() -> {
+                if (isSetupAppendCompleted(segment, writerId)) {
+                    try {
+                        log.trace("Informing writer {} that sent request {}, about token expiry for segment {}",
+                                writerId, requestId, segment);
+                        if (!connection.isClosed()) {
+                            connection.send(new WireCommands.AuthTokenCheckFailed(setupAppend.getRequestId(),
```

Comment:
Yes, that is right. We are depending on the client to meet its contract related to token expiry and reauthenticate. One alternative would have been to close the connection, but that'd be too expensive I suppose.

Reply:
I can maintain a flag in the `WriteState` such that when the token expires, the task will set it to true. Then, I can check the flag on each append. However, that can introduce additional performance cost for each append. I believe the current solution is adequate and provides a good balance of security and performance.

Reply:
This is a question of authentication. You can't rely on the client, that saying "We enforce authorization by asking nicely". The connection needs to be closed.

Reply:
If connection pooling is enabled then closing the connection would cause all the clients using this connection to see a connection drop. But since it is a question of authentication the other clients would see a connection drop and would re-establish a connection with the SSS.

Reply:
I've modified the code to close the connection, as suggested.